test/example-composited.test.js

```diff
@@ -396,7 +470,43 @@ test('single [Array] And Relation failed', t => {
   })
 })
 
-test('[Array] notation & single case Or Relation sucess under first case', t => {
+test('Two sub-arrays Or Relation success', t => {
+  t.plan(2)
+
+  fastify.inject({
+    method: 'POST',
+    url: '/check-two-sub-arrays-or',
+    payload: {
+      n: 11
```

Comment:
I updated `11` to `110` and now `verifyBigAsync` is passing BUT `verifyOddAsync` still called and doesn't wait until `verifyBigAsync` finished

Reply:
@mcollina my thoughts on this — due to broken default behavior it should be 5.0.0, not 4.5.0

Reply:
@isnifer Could you provide a test case for this?

I've just tried the following case locally:
- a route:
```javascript
fastify.route({
  method: 'POST',
  url: '/check-two-sub-arrays-or-2',
  preHandler: fastify.auth([[fastify.verifyBigAsync], [fastify.verifyOddAsync]], { relation: 'or' }),
  handler: (req, reply) => {
    reply.send({
      big: req.big,
      odd: req.odd ?? 'hello',
    })
  }
})
```
- a test:
```javascript
test('Two sub-arrays Or Relation success', t => {
  t.plan(2)

  fastify.inject({
    method: 'POST',
    url: '/check-two-sub-arrays-or-2',
    payload: {
      n: 110
    }
  }, (err, res) => {
    t.error(err)
    const payload = JSON.parse(res.payload)

    t.same(payload, {
      big: true,
      odd: 'hello',
    })
  })
})
```

And the test is passing meaning the `fastify.verifyOddAsync` was not called.

Reply:
@yakovenkodenis yes, sure, I forked this repo https://github.com/isnifer/fastify-auth.

1. I hide all routes except this one — https://github.com/isnifer/fastify-auth/blob/master/test/example-composited.js#L148-L159
2. Added important `console.log()`s here — https://github.com/isnifer/fastify-auth/blob/master/test/example-composited.js#L52-L73
3. And hide all tests except this one — https://github.com/isnifer/fastify-auth/blob/master/test/example-composited.test.js#L473-L490

And my test logs are:
```sh
npm t

> @fastify/auth@4.5.0 test
> npm run test:unit && npm run test:typescript


> @fastify/auth@4.5.0 test:unit
> tap

 PASS ​ ./test/auth.test.js 29 OK 38.833ms
 PASS ​ ./test/example-async.test.js 18 OK 48.952ms
./test/example-composited.test.js 1> If you see it — IT IS OK
./test/example-composited.test.js 1> FIRST_ERROR undefined
./test/example-composited.test.js 1> THIS IS A SECOND AUTH PATH SHOULD NOT BE CALLED
 PASS ​ ./test/example-composited.test.js 2 OK 17.618ms
 PASS ​ ./test/example.test.js 20 OK 50.974ms
```

Reply:
@yakovenkodenis also important note — I've changed line https://github.com/isnifer/fastify-auth/blob/master/test/example-composited.test.js#L481 to make `verifyBigAsync` pass since it's a **first** auth path

Reply:
@Eomm no problem, I'll do it

Reply:
@Eomm I've created #223